internal/ocm/resource/cluster_test.go

```diff
@@ -0,0 +1,186 @@
+		Expect(computeLabels["key1"]).To(Equal("val1"))
+		Expect(ocmClusterNode.AvailabilityZones()).To(BeEmpty())
+		Expect(ocmClusterNode.Compute()).To(Equal(2))
+		autoscaleCompute := ocmClusterNode.AutoscaleCompute()
+		Expect(autoscaleCompute).To(BeNil())
+	})
+	It("Autoscaling disabled multiAZ false set one availability zone - success", func() {
+		err := cluster.CreateNodes(ctx, false, nil, nil, nil, nil, nil, []string{"us-east-1a"}, false)
+		Expect(err).NotTo(HaveOccurred())
+		ocmCluster, err := cluster.Build()
+		Expect(err).NotTo(HaveOccurred())
+		ocmClusterNode := ocmCluster.Nodes()
+		Expect(ocmClusterNode).NotTo(BeNil())
+		Expect(ocmClusterNode.ComputeMachineType()).To(BeNil())
+		Expect(ocmClusterNode.ComputeLabels()).To(BeEmpty())
+		azs := ocmClusterNode.AvailabilityZones()
+		Expect(azs).To(HaveLen(1))
+		Expect(ocmClusterNode.Compute()).To(Equal(2))
+		autoscaleCompute := ocmClusterNode.AutoscaleCompute()
+		Expect(autoscaleCompute).To(BeNil())
+	})
+	It("Autoscaling disabled multiAZ false set three availability zones - failure", func() {
+		err := cluster.CreateNodes(ctx, false, nil, nil, nil, nil, nil, []string{"us-east-1a", "us-east-1b", "us-east-1c"}, false)
+		Expect(err).To(HaveOccurred())
+		Expect(err.Error()).To(Equal("The number of availability zones for a single AZ cluster should be 1, instead received: 3"))
+	})
+	It("Autoscaling disabled multiAZ true set three availability zones and two replicas - failure", func() {
+		err := cluster.CreateNodes(ctx, false, int64Pointer(2), nil, nil, nil, nil, []string{"us-east-1a", "us-east-1b", "us-east-1c"}, true)
+		Expect(err).To(HaveOccurred())
+		Expect(err.Error()).To(Equal("Multi AZ cluster requires at least 3 compute nodes"))
+	})
+	It("Autoscaling disabled multiAZ true set three availability zones and three replicas - success", func() {
+		err := cluster.CreateNodes(ctx, false, int64Pointer(3), nil, nil, nil, nil, []string{"us-east-1a", "us-east-1b", "us-east-1c"}, true)
+		Expect(err).NotTo(HaveOccurred())
+		ocmCluster, err := cluster.Build()
+		Expect(err).NotTo(HaveOccurred())
+		ocmClusterNode := ocmCluster.Nodes()
+		Expect(ocmClusterNode).NotTo(BeNil())
+		Expect(ocmClusterNode.ComputeMachineType()).To(BeNil())
+		Expect(ocmClusterNode.ComputeLabels()).To(BeEmpty())
+		azs := ocmClusterNode.AvailabilityZones()
+		Expect(azs).To(HaveLen(3))
+		Expect(ocmClusterNode.Compute()).To(Equal(3))
+		autoscaleCompute := ocmClusterNode.AutoscaleCompute()
+		Expect(autoscaleCompute).To(BeNil())
+	})
+	It("Autoscaling disabled multiAZ true set one zone - failure", func() {
+		err := cluster.CreateNodes(ctx, false, nil, nil, nil, nil, nil, []string{"us-east-1a", "us-east-1b", "us-east-1c"}, true)
+		Expect(err).To(HaveOccurred())
+		Expect(err.Error()).To(Equal("Multi AZ cluster requires at least 3 compute nodes"))
+	})
+})
+
+func int64Pointer(src int64) *int64 {
+	return &src
+}
+
+func stringPointer(src string) *string {
+	return &src
+}
```

Comment:
```suggestion
func Pointer[T ~int64 | ~string](src T) *T {
	return &src
}
```

Reply:
Changed it to generic of type any
This generic function enforce me to declare that the type is from type int64, otherwise its int
So I called the function like: `pointer(int64(2))`

Reply:
Even better